procrustes/utils.py

```diff
@@ -336,3 +339,55 @@ def _check_arraytypes(*args):
         raise TypeError("Matrix inputs must be NumPy arrays")
     if any(x.ndim != 2 for x in args):
         raise TypeError("Matrix inputs must be 2-dimensional arrays")
+
+
+class OptResult(dict):
+    r"""Represents the Procrustes optimization result.
+
+    Attributes
+    ----------
+    new_a : ndarray
+        The translated/scaled numpy ndarray :math:`\mathbf{A}`.
+    new_b : ndarray
+        The translated/scaled numpy ndarray :math:`\mathbf{B}`.
+    array_u : ndarray
+        The right hand side optimum transformation matrix.
+    array_p : ndarray
+        The left hand side transformation matrix for two-sided Procrustes problem with
+        two transformation.
+    array_q : ndarray
+        The right hand side transformation matrix for two-sided Procrustes problem with
+        two transformation.
+    e_opt : float
+        Two-sided permutation Procrustes error.
+
+    Notes
+    -----
+    Some of additional attributes are not listed above can be foundt at specific Procrustes
+    algorithm implementations. Since this class is essentially a subclass of dict with attribute
+    accessors, one can see which attributes are available using the `keys()` method.
+    """
+
+    # modification on https://github.com/scipy/scipy/blob/v1.4.1/scipy/optimize/optimize.py#L77-L132
+    def __getattr__(self, name):
+        """Deal with attributes which it doesn't explicitly manage."""
+        try:
+            return self[name]
+        except KeyError:
+            raise AttributeError(name)
+
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def __repr__(self):
```

Comment:
"human friendly" should be __str__ method.  __repr__ should be a string that can be put into eval() and return your object.

Reply:
double-underscored;; _ _ str _ _ and _ _ repr _ _.  Github's markdown doesn't like that.

Reply:
> "human friendly" should be **str** method. **repr** should be a string that can be put into eval() and return your object.

I am not sure if I understand it correctly. My guess is you suggest to use `__str__`  (double underscore) instead of `__repr__`. Is this what you suggest? Thanks for pointing it out.

Reply:
> double-underscored;; _ _ str _ _ and _ _ repr _ _. Github's markdown doesn't like that.

But if we use _ _ str _ _ instead of `__str__`, would it be problematic for python?